internal/testkeys/testkeys.go

```diff
@@ -0,0 +1,54 @@
+}
+
+func New[T crypto.PrivateKey](keyString EncodedKeyString) (*T, error) {
```

Comment:
I think this would be easier to use if the signature was 

```go
func New[T crypto.PrivateKey](t *testing.T, keyString EncodedKeyString) *T
```

and the function would fail the test if the key does not decode.